pkg/controller/endpointslice/endpointslice_controller_test.go

```diff
@@ -256,20 +256,28 @@ func TestSyncServiceEndpointSliceLabelSelection(t *testing.T) {
 	numActionsBefore := len(client.Actions()) + 1
 	standardSyncService(t, esController, ns, serviceName, "false")
 
-	if len(client.Actions()) != numActionsBefore+5 {
-		t.Errorf("Expected 5 more actions, got %d", len(client.Actions())-numActionsBefore)
+	// TODO(robscott): Find a way for lister and fake clientset to be more
+	// closely linked. In a real world test, updating the partially matching
+	// EndpointSlice to have the managed-by label would be enough to have it
+	// selected by the lister later in syncService call, but that is not the
```

Comment:
> In a real world test, updating the partially matching
> EndpointSlice to have the managed-by label would be enough to have it
> selected by the lister later in syncService call

It takes time for the update/patch to propagate into the lister cache via a watch event... you cannot update an object and depend on observing the result of that update in the same syncService call. In fact, if you do a later Update of that object based on the cache copy, you are extremely likely to get a Conflict error conflicting with your earlier update.